node/src/components/block_synchronizer/tests.rs

```diff
@@ -336,3 +452,1231 @@ async fn should_not_stall_after_registering_new_era_validator_weights() {
         }) if peers.contains(&peer) && id.block_hash == *block.hash()
     );
 }
+
+#[test]
+fn duplicate_register_block_not_allowed_if_builder_is_not_failed() {
+    let mut rng = TestRng::new();
+    let test_env = TestEnv::random(&mut rng);
+    let block = test_env.block();
+    let validator_matrix = test_env.gen_validator_matrix();
+    let mut block_synchronizer = BlockSynchronizer::new_initialized(&mut rng, validator_matrix);
+
+    // Register block for forward sync
+    assert!(block_synchronizer.register_block_by_hash(*block.hash(), false, true));
```

Comment:
I know it's not a part of this PR, but can you extend [the comment](https://github.com/casper-network/casper-node/blob/cc95de3a80cdada1d67a84a2178f23bcf10e36ae/node/src/components/block_synchronizer.rs#L275) for `register_block_by_hash` and describe what is the meaning of the returned `bool`?